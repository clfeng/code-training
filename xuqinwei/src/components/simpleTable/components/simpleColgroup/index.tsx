import { defineComponent } from 'vue'

export default defineComponent({
  name: 'simpleColgroup',
  props: {
    dataColumns: {
      type: Array,
      default: []
    },

  },
  setup(props) {
    return () => {
      return (
        <colgroup>
          {
            props.dataColumns.map(column => <col key={column.label} width={column.width || 40} />)
          }
        </colgroup>
      )
    }
  },
})
