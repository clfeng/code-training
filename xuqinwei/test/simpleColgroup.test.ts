import { shallowMount } from "@vue/test-utils";
import { expect, test, vi } from "vitest";
import simpleColgroup from "../src/components/simpleTable/components/simpleColgroup";


test("mount component", () => {
  const wrapper = shallowMount(simpleColgroup, {
    props: {
      dataColumns: []
    }
  }
  );

  expect(wrapper.html()).toMatchSnapshot();
});

