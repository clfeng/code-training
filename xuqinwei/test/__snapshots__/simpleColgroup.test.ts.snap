// Vitest Snapshot v1

exports[`mount component 1`] = `"<colgroup></colgroup>"`;
